### PR TITLE
Registry Action

### PR DESCRIPTION
**Pull request by nebhale**
This change adds a GitHub action that facilities adding and yanking buildpacks from the registry index.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM golang:latest as build-stage
```
Comment (jkutner) [inline Dockerfile:1]: should we pin to a 1.15 image to match the version in `go.mod`, or do we expect `latest` to support 1.15 forever?

Comment (nebhale) [reply Dockerfile:1]: I go the other way on this and say that our code will support Golang `latest` forever. But given Dependabot keeping up with that for us, I'll update to 1.15.

**Review by jkutner — APPROVED** [review]

File: .github/workflows/create-action-registry.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        runs-on:
+            - ubuntu-latest
+        steps:
+            - if: ${{ github.event_name != 'pull_request' || ! github.event.pull_request.head.repo.fork }}
+              name: Docker login ghcr.io
+              uses: docker/login-action@v1
+              with:
+                password: ${{ secrets.IMPLEMENTATION_GHCR_TOKEN }}
+                registry: ghcr.io
+                username: ${{ secrets.IMPLEMENTATION_GHCR_USERNAME }}
```
Comment (jkutner) [inline .github/workflows/create-action-registry.yml:26]: where do these `IMPLEMENTATION_GHCR_*` secrets come from? Is this a standard GHA thing?

Comment (nebhale) [reply .github/workflows/create-action-registry.yml:26]: They're secrets in that I've [added to our org](https://github.com/organizations/buildpacks/settings/secrets).  The could've been named anything, but GHCR tokens for the implementation team in this case.

File: README.md [hunk 3]
```diff
@@ -3,6 +3,45 @@
...
+
+[bri]: https://github.com/buildpacks/registry-index
+
+### Add
+```yaml
+uses: docker://ghcr.io/buildpacks/actions/registry
+with:
+  token:   ${{ secrets.IMPLEMENTATION_PAT }}
+  id:      $buildpacksio/test-buildpack
+  version: {{ steps.deploy.outputs.version }}
```
Comment (jkutner) [inline README.md:17]:
```suggestion
  version: ${{ steps.deploy.outputs.version }}
```

**Review by jkutner — APPROVED** [review]

**Pull request merged**